File: Go/models/todo.go
package models

import (
	"time"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Todo struct {
	Id       primitive.ObjectID 				`json:"id,omitempty"`
	Title    string             				`json:"title,omitempty"`
	Content  string             				`json:"content,omitempty"`
	Priority string             				`json:"priority,omitempty"`
	UserId   primitive.ObjectID                 `json:"userId,omitempty"`
	GroupId  primitive.ObjectID                 `json:"groupId,omitempty"`
	DueDate  time.Time          				`json:"dueDate,omitempty"`
}

File: vue-client/src/classes/todo.ts
import {Entity} from "@/interfaces/entity";

export class Todo implements Entity {
    id: number;
    title: string;
    content: string;
    userId: string;
    priority: string;
    groupId: string;


    constructor(id = 0,title='', content = '', userId = '',priority='',groupId='') {
        this.id = id;
        this.title = title;
        this.userId = userId;
        this.priority=priority;
        this.groupId=groupId;
        this.content=content;
    }
}